web/html/add_test.html

```diff
@@ -693,5 +758,131 @@ <h2 class="mt-4 text-lg font-semibold">Marking Scheme</h2>
         });
         return marks;
     }
+
+    async function validateAndCreateTest(data) {
+        const testRule = data.TestRule || {};
+
+        const questionsTableBody = document.getElementById("questions-table-body");
+        const questionRows = questionsTableBody.querySelectorAll("tr[id^='problem-']");
+        const subjectRows = questionsTableBody.querySelectorAll("tr[id^='subject-']");
+        const subtypeRows = questionsTableBody.querySelectorAll("tr[id^='subtype-']");
```

Comment:
subtype should be counted "within" subject, right? there would be subsections within each subject?

so physics - 2, chem - 2.... but the subtyperows here will be 4 ?

Reply:
but i feel this may select subtype across subjects